### ISSUE TITLE
Allow image upload in chat

### ISSUE THREAD

**Issue by cameronurnes** [owncast/owncast#3714]
### Share your bug report, feature request, or comment.

Another thing my users ask for/miss when not using kosmi is the ability to send jpg images and gifs in the chat window.  Example photo attached.

<img width="293" alt="Screenshot 2024-04-21 at 3 57 11 PM" src="https://github.com/owncast/owncast/assets/8864171/aa33494a-6c18-410f-9576-209b1b00b2ef">

**Comment by gabek** [comment]
I understand there are probably a ton of features other chat platforms have. But in all honesty, I don't think of Owncast as a chat platform. It has chat, and when enough people ask for a specific feature long enough then new things get in, but continuous improvement of the chat is generally not my priority. 

So if we were to add image uploading, it would add a new layer of image moderation, and disk space management. For some historical background, chat participants used to be able to link to images in chat and they would render inline. Nobody really used it, and the only people who did, used it to harass other people.

There's a lot of super exciting stuff on the [roadmap](https://owncast.online/roadmap) right now, but frontend chat features really aren't any of them. 

As always, my disclaimer is that if somebody is really passionate about this space and wanted to spearhead this feature and help maintain it into the future as well, I'm absolutely up for a conversation. And I'm also not saying no, but there isn't anything planned for anything like this right now.

I'll keep this issue open for others to chime in if they have ideas on how this could work safely that didn't just fill up Owncast streamer's drives with dick pics.

**Comment by cameronurnes** [comment]
If it might be easier, a checkbox to enable that inline feature again would probably suffice.  We are a private group so we have no harassment issues, but we still get 10 to 20 people per night and it's a common request when I use owncast's website.

**Comment by gabek** [comment]
I understand you're thinking about your one group, but I'm thinking about all Owncast users and if it will
1. Benefit streamers and their streams. Do they want it? Will they use it?
2. Is it worth it to the project? Is it hard to maintain? Is it hard to support? Is it more trouble than it is worth? Does it satisfy the goals outlined in the [product definition document](https://github.com/owncast/owncast/blob/develop/docs/product-definition.md)?

**Comment by JeremyMahieu** [comment]
Yeah the custom emoji are already too much imo. Maybe integrate with some other already existing chat service.

**Comment by germainelee** [comment]
> For some historical background, chat participants used to be able to link to images in chat and they would render inline. Nobody really used it, and the only people who did, used it to harass other people.

Let's say we have the image upload/link feature, I'm thinking that the admin would need to review and approve that message first before the image would appear in the chat. And the image message would be disappearing media where it will be removed after a few seconds.

Would this work?

**Comment by cameronurnes** [comment]
In our group we wouldn't need the approvals, so would want a checkbox on that feature too (leave it up to the server owner etc)

**Comment by gabek** [comment]
I completely understand that there are specific use cases for this request. But I'm looking at it from a high level. There are a ton of different uses for Owncast. TV networks, event venues, churches, etc etc etc. None of these use cases would reasonably benefit from this feature. A group of people streaming to their friends, however, I could see absolutely would.

Like always, this isn't to say that it would never get done, but we do have a roadmap of things that are currently planned and expect to be beneficial for a wide variety of Owncast users. Building things that benefit the most number of people is really the goal with the limited resources we have.

**Comment by germainelee** [comment]
Not stale. Should be marked as backlog. See above comment by maintainer
> Like always, this isn't to say that it would never get done

**Comment by gabek** [comment]
The administration required for this would be a lot of work, and I still believe that only a small sliver of servers would use it.

> TV networks, event venues, churches, etc etc etc. None of these use cases would reasonably benefit from this feature. 

> So if we were to add image uploading, it would add a new layer of image moderation, and disk space management. For some historical background, chat participants used to be able to link to images in chat and they would render inline. Nobody really used it, and the only people who did, used it to harass other people.

**Comment by cameronurnes** [comment]
If it used to do in-line rendering and it was already working, could that be re-integrated?  Just have it behind a checkbox to enable?  Definitely understand why this is a low priority though.

**Comment by gabek** [comment]
It was only supported on the previous version of the web interface. [It supported](https://github.com/owncast/owncast/commit/777de9fbf4f46a50adbc5fb5f3999d0abd407509#diff-81317a066a11e947476350fc387c8cd2f1c4890ac4b0c90fdc6f834363aa7651L131) direct image URLs, Instagram links, and YouTube links (embedded video in the chat). 

If this was something a lot of people wanted, I wouldn't be opposed to adding it, and defaulting it to off. But nobody noticed when these were no longer supported.
